### PR TITLE
fix(task): support multi-line arrays in file task headers

### PR DESCRIPTION
**Pull request by Marukome0743**
## Problem

Each `#MISE` header line in a file task is parsed as an independent TOML document, so a value cannot be split across lines:

```bash
#!/usr/bin/env bash
#MISE depends=[
#MISE   "lint",
#MISE ]
echo hi
```

This is not merely ignored — the `depends=[` fragment parses as unterminated TOML and the error propagates out of task loading, so `mise tasks ls` fails for the **whole project**:

```
mise ERROR failed to parse task header TOML "depends=[": TOML parse error at line 1, column 10
  |
1 | depends=[
  |          ^
unclosed array, expected `]`
```

One task's formatting choice breaks every task in the project.

## Change

- Scan header lines into logical entries. An entry starts on a `key=value` line and, while it leaves a TOML array, inline table or multi-line string open, absorbs the following header lines. The set of lines that can *start* an entry is unchanged, so any header that parses today produces the identical value and the identical error text.
- Exclude continuation lines from `extract_usage_from_comments`. They don't look like `key=value`, so they would otherwise have been handed to the usage spec parser as spec text.
- Decode `file_has_decoded_template` per entry, so one unparseable header can no longer hide a decoded template in another entry. This gates trust in the fail-closed direction.
- Report multi-line entries with their line range rather than quoting only the first fragment.
- Document multi-line values in `docs/tasks/file-tasks.md`, which previously did not mention them.

Bracket counting ignores brackets inside basic/literal/multi-line strings and after `#` comments, so `#MISE description="see [1]"` does not open a continuation.

Since headers are parsed with `toml` 1.x (TOML 1.1), inline tables may span lines as well as arrays; both are covered by tests.

This is the last outstanding item from the linked discussion: spaces around `=` already work, and scalar `depends="x"` was fixed by #11382.

## Testing

- `cargo fmt --all -- --check`
- `shellcheck -x e2e/tasks/test_task_file_header_multiline`
- `shfmt -d -s e2e/tasks/test_task_file_header_multiline`
- `prettier --check docs/tasks/file-tasks.md`
- `markdownlint docs/tasks/file-tasks.md`

Added coverage:

- unit tests for the scanner (entry grouping, brackets inside strings), for loading multi-line `depends`/`sources`, for an array of inline tables spanning lines, for a multi-line inline table, for the unterminated-array error message, for usage-spec extraction, and for template detection across entries
- `e2e/tasks/test_task_file_header_multiline`

This machine has 3.7 GB of RAM and no swap, so a cold full-dependency build is not viable here and CI is the first full type-check and test run. To compensate, the scanner and `parse_mise_header_toml` were extracted verbatim into a standalone crate pinned to the same `toml` version and exercised against every scenario above, and the format-dependent e2e assertions were validated against a released `mise` binary.

Addresses https://github.com/jdx/mise/discussions/4603

*AI-assisted — Tool: Claude Code; model: anthropic/claude-opus-5; version: 2.1.220.*

## Summary by CodeRabbit

* **New Features**
  * Task file `#MISE` headers now support multi-line TOML values, including multi-line arrays and arrays of inline tables.
* **Bug Fixes**
  * Multi-line `#MISE` continuations are no longer misinterpreted as usage text.
  * Improved parsing robustness and clearer error reporting for invalid/incomplete multi-line headers, with accurate line ranges; unparseable entries no longer block other valid header values.
* **Documentation**
  * Added guidance and examples for multi-line TOML in `#MISE` comments, including dotted-key workarounds.
* **Tests**
  * Added end-to-end coverage for multiline header parsing and edge cases (including inline-table and dotted-key scenarios).

### CODE REVIEW

File: src/task/mod.rs [hunk 1]
```diff
@@ -669,18 +669,194 @@ pub struct Task {
...
+    /// Feed a single line of TOML.
+    fn feed(&mut self, line: &str) {
+        let b = line.as_bytes();
+        let mut i = 0;
+        while i < b.len() {
+            let c = b[i];
+            if let Some(q) = self.multiline {
+                if c == q && b[i + 1..].starts_with(&[q, q]) {
+                    self.multiline = None;
+                    i += 3;
```
[the remark this answers is not in the document]
Comment (Marukome0743) [reply src/task/mod.rs]: Good catch — this is real, and fixed in `df2c1b1`.

Inside a multi-line basic string, `\"""` is an escaped quote followed by two more, not the closing delimiter. `feed` didn't honor `\` escapes, so it closed the string one byte early, ended the entry on that line, and the truncated fragment then failed to parse.

`feed` now skips escapes while inside a multi-line **basic** string. Literal (`'''`) strings have no escapes, so the branch is guarded on the quote byte:

```rust
if q == b'"' && c == b'\\' {
    i += 2;
} else if c == q && b[i + 1..].starts_with(&[q, q]) {
    self.multiline = None;
    i += 3;
}
```

Verified against the case you describe:

```
#MISE description="""abc \""" def
#MISE ghi"""
```

before: entry spans line 1 only, parse fails
after: entry spans lines 1-2, parses to `abc """ def\nghi`

Added `test_scan_mise_header_entries_multiline_basic_string` to pin it down.

*AI-assisted — Tool: Claude Code; model: anthropic/claude-opus-5; version: 2.1.220.*

File: src/task/mod.rs [hunk 2]
```diff
@@ -669,18 +669,170 @@ pub struct Task {
...
+            } else {
+                entries.push(entry);
+            }
+        }
+    }
+    if let Some((entry, _)) = open {
+        entries.push(entry);
+    }
+    entries
+}
```
[the remark this answers is not in the document]
Comment (Marukome0743) [reply src/task/mod.rs:833]: Not applying this — the premise is incorrect on two counts.

**`# MISE` was never accepted, so nothing is being broken.** The regex this PR replaces was:

```
^(?:#|//|::)(?:MISE| ?\[MISE\]) ([a-z0-9_.-]+\s*=\s*[^\n]+)$
```

The alternation is `MISE` or ` ?\[MISE\]` — a bare space followed by `MISE` matches neither. `# MISE alias="b"` did not match before this PR and does not match after it; the set of lines that can start an entry is deliberately byte-for-byte unchanged.

**This is documented, intentional behavior**, not an oversight ([docs/tasks/file-tasks.md](https://github.com/jdx/mise/blob/main/docs/tasks/file-tasks.md)):

> Beware of formatters changing `#MISE` to `# MISE`. It's intentionally ignored by mise to avoid unintentional configuration. To workaround this, use the alternative: `# [MISE]`.

The proposed `\s*` would make formatter-mangled headers take effect, which is exactly what that rule exists to prevent.

**`test_from_path_toml_headers` passes.** It passed on this branch in CI before the fix commit (`2067 passed; 1 failed`, and the one failure was an unrelated test of mine). It passes because `impl PartialEq for Task` compares only `name`, `args` and `env` — its `alias`/`description`/`sources` assertions are vacuous today. That's worth fixing, but it's pre-existing and out of scope here.

Confirmed empirically against the current implementation: `# MISE alias="b"` yields 0 header entries, matching pre-PR behavior.

On the separate nitpick about the inline-table hint citing `tools.node` — that code no longer exists. mise parses headers with `toml` 1.x (TOML 1.1), which permits multi-line inline tables, so the hint was removed and replaced with a passing test.

*AI-assisted — Tool: Claude Code; model: anthropic/claude-opus-5; version: 2.1.220.*
